### PR TITLE
Create dns_dynu.sh

### PR DESCRIPTION
**Pull request by shar0119**
Add DNS API support for Dynu.

### CODE REVIEW

File: dnsapi/dns_dynu.sh [hunk 1]
```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env sh
+
+my_dir="$(dirname "$0")"
+source "$my_dir/acme.sh"
```
Comment (Neilpang) [inline dnsapi/dns_dynu.sh]: you don't need to `source acme.sh`
remove it.

**Comment by Neilpang** [comment]
Please fix the format errors:　https://travis-ci.org/Neilpang/acme.sh/jobs/211902415

**Comment by tavinus** [comment]
Hi!
To be able to use this, I need to pay for the Dynu API right?

**Comment by shar0119** [comment]
@tavinus If you let us know your Dynu account username, we can give you a set of free API credentials.

**Comment by Neilpang** [comment]
@shar0119 
Please fix the format errors:　https://travis-ci.org/Neilpang/acme.sh/jobs/211902415

File: dnsapi/dns_dynu.sh [hunk 2]
```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env sh
+
+#Client ID
+Dynu_ClientId="0b71cae7-a099-4f6b-8ddf-94571cdb760d"
```
Comment (Neilpang) [inline dnsapi/dns_dynu.sh]: comment out this line

File: dnsapi/dns_dynu.sh [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+#
+#Secret
+Dynu_Secret="aCUEY4BDCV45KI8CSIC3sp2LKQ9"
```
Comment (Neilpang) [inline dnsapi/dns_dynu.sh]: and comment out this line

Comment (shar0119) [inline dnsapi/dns_dynu.sh]: Commented out as requested.

**Comment by shar0119** [comment]
Could you please merge the script to https://github.com/Neilpang/acme.sh/tree/dev/dnsapi? Do we need to provide documentation before you can do this? Thanks~~

**Comment by Neilpang** [comment]
@shar0119 

Please update the 2 readme.md files:
https://github.com/Neilpang/acme.sh/blob/master/README.md

**Comment by shar0119** [comment]
Two README.md files have been updated.

**Comment by Neilpang** [comment]
Hi @shar0119 
I only see you updated one readme.md file.
Please update the main readme file too: https://github.com/Neilpang/acme.sh/blob/master/README.md
List your api in the supported list.

Thanks.

File: dnsapi/dns_dynu.sh [hunk 4]
```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env sh
+
+#Client ID
+#Dynu_ClientId="0b71cae7-a099-4f6b-8ddf-94571cdb760d"
+#
+#Secret
+#Dynu_Secret="aCUEY4BDCV45KI8CSIC3sp2LKQ9"
+#
```
Comment (Neilpang) [inline dnsapi/dns_dynu.sh]: please add your name and link here, so that one can report bugs to you.
Thanks.

File: dnsapi/dns_dynu.sh [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+    _err "Could not add TXT record"
+    return 1
+  fi
+
+  return 0
+}
+
+#fulldomain
+dns_dynu_rm() {
+  fulldomain=$1
```
Comment (Neilpang) [inline dnsapi/dns_dynu.sh]: Is it possible for you to implement the `rm` function ?

Comment (shar0119) [inline dnsapi/dns_dynu.sh]: Sure. We will work on it. Thanks!

Comment (Neilpang) [inline dnsapi/dns_dynu.sh]: Cool

Comment (shar0119) [inline dnsapi/dns_dynu.sh]: We have implemented the rm function. Thanks!

**Comment by shar0119** [comment]
We have made the change regarding _egrep_o() function. Thanks.

**Comment by shar0119** [comment]
Could you please merge the script to https://github.com/Neilpang/acme.sh/tree/dev/dnsapi? Please do let us know if anything is required from our side. Thanks.

**Comment by Neilpang** [comment]
sorry, I was too busy recently.

**Comment by Neilpang** [comment]
@shar0119 
Hi,
I just noticed there is still a `grep -Po`:

https://github.com/Neilpang/acme.sh/blob/dev/dnsapi/dns_dynu.sh#L152

Can you please fix it ?

**Pull request merged**